ext/standard/pack.c

```diff
@@ -885,12 +885,15 @@ PHP_FUNCTION(unpack)
 			if ((inputpos + size) <= inputlen) {
 
 				zend_string* real_name;
+				zend_long long_key = 0;
 				zval val;
 
 				if (repetitions == 1 && namelen > 0) {
 					/* Use a part of the formatarg argument directly as the name. */
 					real_name = zend_string_init_fast(name, namelen);
-
+				} else if (namelen == 0) {
+					real_name = NULL;
+					long_key = i + 1;
```

Comment:
GH is not letting me do a single suggested change across "deleted lines"...

But if you reorder the ifs you don't need to check the `namelen` value again, as you know it will be `> 0` (well unless you tell me this can somehow be negative...)
```suggestion
				if (namelen == 0) {
					real_name = NULL;
					long_key = i + 1;
				} else if (repetitions == 1) {
					/* Use a part of the formatarg argument directly as the name. */
					real_name = zend_string_init_fast(name, namelen);
```